meg_starting_kit.ipynb

```diff
@@ -1025,97 +1055,77 @@
-    "import glob\n",
-    "\n",
-    "data_dir = 'data/'\n",
-    "\n",
-    "# find all the files ending with '_lead_field' in the data directory\n",
-    "lead_field_files = os.path.join(data_dir, '*lead_field.npz')\n",
-    "lead_field_files = sorted(glob.glob(lead_field_files))\n",
-    "\n",
-    "parcel_indices_leadfield, L = [], []\n",
-    "subj_dict = {}\n",
-    "for idx, lead_file in enumerate(lead_field_files):\n",
-    "    lead_matrix = np.load(lead_file)\n",
+    "from problem import get_leadfields\n",
     "\n",
-    "    lead_file = os.path.basename(lead_file)\n",
-    "    subj_dict['subject_' + lead_file.split('_')[1]] = idx\n",
-    "\n",
-    "    parcel_indices_leadfield.append(lead_matrix['parcel_indices'])\n",
-    "\n",
-    "    # scale L to avoid tiny numbers\n",
-    "    L.append(1e8 * lead_matrix['lead_field'])\n",
-    "    assert parcel_indices_leadfield[idx].shape[0] == L[idx].shape[1]\n",
-    "\n",
-    "assert len(parcel_indices_leadfield) == len(L) == idx + 1\n",
-    "assert len(subj_dict) >= 1  # at least a single subject\n",
-    "\n",
-    "print(f'Loaded {len(L)} lead_fields and {len(parcel_indices_leadfield)} parcel_indices')\n",
-    "print(f'Created dictionary of subject_ids and matching indices: {subj_dict}')"
+    "Ls, parcel_indices = get_leadfields()\n",
+    "print(f'Loaded {len(Ls)} lead_fields and {len(parcel_indices)} parcel_indices')\n",
+    "print(f'List of subject_ids: {list(Ls.keys())}')"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "We created the `subj_dict` to keep track which row of `L['lead_field']` and which row of `parcel_indices_leadfield` correspond to which subject."
+    "How can we use this information for the predictions?"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "Now we will use `subj_dict` to map the subjects in the X datasets:"
+    "## Lasso Lars algorithm <a class=\"anchor\" id=\"lassolars\"></a> "
    ]
   },
   {
-   "cell_type": "code",
-   "execution_count": 30,
+   "cell_type": "markdown",
    "metadata": {},
-   "outputs": [],
    "source": [
-    "X_train_mapped = X_train.copy()\n",
-    "X_train_mapped['subject_id'] = X_train['subject'].map(subj_dict)\n",
-    "# scale to avoid tiny numbers\n",
-    "X_train_mapped.iloc[:, :-2] *= 1e12\n",
+    "We will now construct slightly more complicated estimator which will use 'lead_fields'. First we want to load the 'lead_fields' which are used in our data.\n",
+    "\n",
+    "*Remark:* Note that we are scaling all the 'lead_fields' by 1e8 in the `get_leadfields` function. That is to avoid having too small numbers given to the estimator.\n",
     "\n",
-    "X_test_mapped = X_test.copy()\n",
-    "X_test_mapped['subject_id'] = X_test_mapped['subject'].map(subj_dict)\n",
-    "# scale to avoid tiny numbers\n",
-    "X_test_mapped.iloc[:, :-2] *= 1e12"
+    "The idea of the Lasso is estimate $z \\in \\mathbb{R}^q$ as:\n",
```

Comment:
'is to solve the inverse problem, estimate...'

Maybe talk about the inverse problem here, as it's confusing when you mention it below, as people might think you are talking about the inverse of this equation